Extension/src/SSH/commands.ts

```diff
@@ -0,0 +1,53 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+import { ISshHostInfo, ProcessReturnType } from '../common';
+import { defaultSystemInteractor } from './commandInteractors';
+import { runSshTerminalCommandWithLogin } from './sshCommandRunner';
+
+export async function scp(files: vscode.Uri[], host: ISshHostInfo, targetDir: string, jumpHosts?: ISshHostInfo[], cancellationToken?: vscode.CancellationToken): Promise<ProcessReturnType> {
+    const args: string[] = [];
+    if (jumpHosts && jumpHosts.length > 0) {
+        args.push('-J', jumpHosts.map(getFullHostAddress).join(','));
+    }
+    if (host.port) {
+        // upper case P
+        args.push('-P', `${host.port}`);
+    }
+    args.push(files.map(uri => `"${uri.fsPath}"`).join(' '), `${getFullHostAddressNoPort(host)}:${targetDir}`);
+
+    return runSshTerminalCommandWithLogin(host, { systemInteractor: defaultSystemInteractor, nickname: 'scp', command: `scp ${args.join(' ')}`, token: cancellationToken });
```

Comment:
Does this require the user to have `scp` in their `PATH`?

For `pipeTransport` we allow users to provide the full path using `pipeProgram`.

I can see this being an issue on Windows 7 and Windows 10 if they do not have OpenSSH installed. 
Do we want to add a way for users to customize the path to the programs?

Reply:
This feature already requires a rather recent version of VS Code because of the proposed API we are using. I would be surprised if the users don't have ssh/scp on their path. I think it's safe enough to see if we receive any related feedbacks. However, if you think this is a concerning issue, I think adding an entry to user/machine settings makes more sense, since launch.json is supposed to be shared across a team.

Reply:
VS Code still has support for Windows 8, so those users may still have to point to an external path for SSH / SCP

Reply:
We may need to support rsync (probably others?) in the future. Adding one setting entry for each path seems quite awkward. @bobbrow Any suggestions?

Reply:
I don't find that awkward.  The property can be optional for those who want to set an explicit path.  There is already precedent for it with `pipeProgram`.

Reply:
@bobbrow I prefer adding it to user/machine settings than launch.json, since launch.json should be shared in a team, and each team member may install them at different locations.

Reply:
How is it different from the `${targetDir}` variable in your example?  Team members may prefer different output sites as well.  The way to make it work for cross-team configurations is to use environment variables.  My opinion is that we shouldn't add settings to the C++ extension that are essentially just environment variables for launch configurations.

Reply:
TargetDir is usually fixed, just like default install dir for software, especially for embedded devices which are re-imaged all the time. Also I think adding to env var is hardly different from adding to path... So how about not adding this at all and just assume they are on path? :D

Reply:
It's usually a good idea to provide a way to override the path to a given tool, but I don't have a strong opinion about it.  If @WardenGnaw doesn't object, you can leave it out and see if you get feedback from developers who use the feature.

I only have a strong opinion favoring adding a property to the launch configuration instead of adding a new setting.

Reply:
Discussed offline. I would prefer a path override defined.